pycheribuild/projects/cross/ros2.py

```diff
@@ -0,0 +1,99 @@
+#    documentation and/or other materials provided with the distribution.
+#
+# THIS SOFTWARE IS PROVIDED BY THE AUTHOR AND CONTRIBUTORS ``AS IS'' AND
+# ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+# IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+# ARE DISCLAIMED.  IN NO EVENT SHALL THE AUTHOR OR CONTRIBUTORS BE LIABLE
+# FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL
+# DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS
+# OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION)
+# HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT
+# LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY
+# OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF
+# SUCH DAMAGE.
+#
+
+from os.path import isdir
+from .crosscompileproject import *
+# from ..project import ReuseOtherProjectRepository
+
+
+class BuildRos2(CrossCompileCMakeProject):
+    project_name = "ros2"
+    repository = GitRepository("https://github.com/dodsonmg/ros2_dashing_minimal.git", default_branch="master", force_branch=True)
+
+    # atm, we build and install in the sourceDir.
+    # it may eventually be useful to install to rootfs or sysroot depending on whether we want to use ROS2
+    # as a library for building other applications using cheribuild
+    # therefore, these _install_dir don't do anything, but cheribuild requires them
+    native_install_dir = DefaultInstallDir.IN_BUILD_DIRECTORY
+    cross_install_dir = DefaultInstallDir.ROOTFS
+
+    dependencies = ["poco"]
+
+    def _ignore_packages(self):
+        packages = ["src/ros2/rcl_logging/rcl_logging_log4cxx"] # relative to self.sourceDir
+        for package in packages:
+            cmdline = ["touch"]
+            cmdline.append(str(self.sourceDir / package / "COLCON_IGNORE"))
+            self.run_cmd(cmdline, cwd=self.sourceDir)
+        return
+
+    def _make_src_dir(self):
+        cmdline = ["mkdir", self.sourceDir / "src"]
+        return self.run_cmd(cmdline, cwd=self.sourceDir)
+
+    def _run_vcs(self):
+        cmdline = ["vcs", "import", "--input", "ros2_minimal.repos", "src"]
+        return self.run_cmd(cmdline, cwd=self.sourceDir)
+
+    def _run_colcon(self, **kwargs):
+        colcon_cmd = ["colcon", "build"]
+        colcon_args = ["--no-warn-unused-cli"]
+        cmake_args = ["--cmake-args"]
+        cmake_args.append("-DBUILD_TESTING=NO")
+        cmake_args.append("-DCMAKE_TOOLCHAIN_FILE=" + str(self.sourceDir / "CrossToolchain.cmake"))
+        cmdline = colcon_cmd + cmake_args + colcon_args
+        if self.config.verbose:
+            cmdline.append("--event-handlers")
+            cmdline.append("console_cohesion+")
+        return self.run_cmd(cmdline, cwd=self.sourceDir, **kwargs)
+    
+    def update(self):
+        super().update()
+        if not isdir(self.sourceDir / "src"):
```

Comment:
`if not (self.sourceDir / "src").is_dir():`

Reply:
`if not isdir(str(self.sourceDir / "src"))`?

It doesn't seem to want to take a `Posix.path` and apply `isdir()`.